### PR TITLE
Initial documentation for ADL-S

### PR DESCRIPTION
**Pull request by kvparikh89**
Adding initial documentation for ADLS RVP.
Minor changes to TGL platform.

Signed-off-by: Kalp Parikh <kalp.parikh@intel.com>

### CODE REVIEW

File: source/substitutions.txt [hunk 1]
```diff
@@ -33,13 +33,16 @@
 
 .. |TGL| replace:: Tiger Lake
 
-.. |DDR4| replace:: Tiger Lake UP3 DDR4
```
Comment (raghavag) [inline source/substitutions.txt:36]: Difference shows these are removed, is that correct? if yes, why do we have to remove these?

File: source/supported-hardware/adls-rvp.rst [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+Alder Lake-S RVP Board
+-----------------------
+
+.. note:: 12th Generation Intel\ |reg| Core\ |trade| Processor, formally known as |ADLS| family.
+
+Supported Boards
+^^^^^^^^^^^^^^^^^^^^^
+
+|SPN| supports **ADL DDR4, SDDR4, DDR5 and SDDR5** platforms.
```
Comment (raghavag) [inline source/supported-hardware/adls-rvp.rst:11]: We discussed lets not mention SODIMM boards at the moment

**Review by raghavag — CHANGES_REQUESTED** [review]
Please address minor comments

File: source/substitutions.txt (hunk 1 above)
Comment (kvparikh89) [reply source/substitutions.txt:36]: Yes this is intentional. I believe DDR4 should not be used as a substitution for TGL UP3 DD4 board. Hence I expanded these where used in source/supported-hardware/tgl-rvp.rst

**Review by raghavag — APPROVED** [review]
looks good to me

**Pull request merged**